src/app/DrawingApp.ts

```diff
@@ -31,28 +33,45 @@ export class DrawingApp {
   }
 
   private addEventListeners() {
-    window.addEventListener("mousemove", (e) => this.handleMouseMove(e))
-    this.canvas.addEventListener("click", (e) => this.handleClick(e))
+    this.canvas.addEventListener("mousedown", (e) => this.startDrawing(e))
+    this.canvas.addEventListener("mouseup", () => this.stopDrawing())
+    this.canvas.addEventListener("mouseout", () => this.stopDrawing())
+    this.canvas.addEventListener("mousemove", (e) => this.handleMouseMove(e))
   }
 
-  private handleClick(event: MouseEvent) {
-    const { x, y } = this.getMousePosition(event)
+  private startDrawing(event: MouseEvent) {
+    this.isDrawing = true
+    this.addPoint(event.clientX, event.clientY)
+  }
 
-    this.engine.color.setForeground(new Color(255, 0, 0))
-    this.engine.drawLine(getPointsToDrawRectangle(x, y, 20))
-    this.engine.color.setForeground(new Color(255, 255, 0))
-    this.engine.drawLine(getPointsToDrawRectangle(x, y, 30))
-    this.engine.color.setForeground(new Color(0, 255, 0))
-    this.engine.drawLine(getPointsToDrawRectangle(x, y, 40))
+  private stopDrawing() {
+    this.isDrawing = false
   }
 
   private handleMouseMove(event: MouseEvent) {
-    const { x, y } = this.getMousePosition(event)
+    if (!this.isDrawing) {
+      const cursorPoint = this.getMousePosition(event)
+      this.drawCursor(cursorPoint.x, cursorPoint.y)
+      return
+    }
+
+    const currentPoint = this.getMousePosition(event)
+    this.currentSegment.push(currentPoint.x, currentPoint.y)
+    this.engine.updateDrawing(this.currentSegment)
```

Comment:
I thiink I'll want some of this logic to be moved into the `DrawingEngine`, I'm imagining something like this `DrawingApp` class is just responsible for context needed for drawing on the web - so maybe a `WebDrawingApp` would be a better name for it?

I'll have to think about that a bit more about the responsibility for each of these. I suspect that the engine ends up with a context for the line segment currently being drawn, at the very least.

As far as this PR goes, probably best to leave it as is so I can sort some of my vague vision here, but here's some stream of thought ✨